discovery_child_development/utils/google_utils.py

```diff
@@ -79,3 +84,125 @@ def create_client() -> bigquery.Client:
     # Initialize a client with the provided credentials
     client = bigquery.Client(credentials=credentials)
     return client
+
+
+def write_like_condition(term: Union[str, List[str]], table: str, field: str) -> str:
+    """Create a LIKE condition for a search term or a list of search terms"""
+    if len(term) == 1:
+        return f'{table}.{field} LIKE "%{term[0]}%"'
+    else:
+        # Write an AND condition if more than one search term
+        return "(" + " AND ".join([f'{table}.{field} LIKE "%{t}%"' for t in term]) + ")"
+
+
+def create_patents_query(search_terms: List[str]) -> str:
+    """Create a query to fetch data from BigQuery using search terms:
+    the query checks search terms in the title and abstract
+
+    Args:
+    search_terms (List[str]): list of search terms
+
+    Returns
+    str: query to fetch data from BigQuery
+    """
+
+    # Create a list of 'LIKE' conditions for each search term
+    like_conditions = {}
+    for field in ["title", "abstract"]:
+        like_conditions[field] = [
+            write_like_condition(term, "gpr", field) for term in search_terms
+        ]
+
+    # Join conditions with 'OR'
+    combined_conditions_title = " OR ".join(like_conditions["title"])
+    combined_conditions_abstract = " OR ".join(like_conditions["abstract"])
+
+    q = f"""
+    WITH
+    pubs as (
+        SELECT DISTINCT
+            pub.publication_number
+        FROM `patents-public-data.patents.publications` pub
+            INNER JOIN `patents-public-data.google_patents_research.publications` gpr ON
+            pub.publication_number = gpr.publication_number
+        WHERE
```

Comment:
I think you need an extra set of parentheses here? I think at the moment there's a risk that the date range restriction only applies to the abstract condition, not the title condition. Putting both the title and abstract conditions (both sides of the OR) in parentheses should fix this.

```suggestion
        WHERE
            (({combined_conditions_title})
            OR ({combined_conditions_abstract}))
            AND pub.grant_date BETWEEN 20190101 AND 20231231
```

Reply:
See also [the order in which operators are evaluated](https://cloud.google.com/bigquery/docs/reference/standard-sql/operators)

Reply:
Good catch, thank you!